src/main/java/Entities/CollaborativeTask.java

```diff
@@ -0,0 +1,217 @@
+package Entities;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class CollaborativeTask extends Task implements Timeblockable{
+    private boolean recurring;
+    private String frequency;
+    private LocalDateTime startTime;
+    private LocalDateTime endTime;
+    private ArrayList<StudentUser> teammates;
+    private ArrayList<StudentUser> pendingTeammates;
+    private ArrayList<StudentUser> declinedTeammates;
+    private StudentUser leader;
+
```

Comment:
For collaborative scheduling a deadline instance variable is needed as well. So like: private LocalDateTime deadline. This would also need a getter method and I think you have to add it to your constructor since the deadline can't have a default value.

Reply:
Added two new constructor and getter/setter methods for new attribute.

Reply:
Would you ever need a Collaborative "Test" or "Assignment"? I am thinking of making CollaborativeEvent a subclass of Event and CollaborativeAssignment a subclass of Assignment. Let me know what you think of that idea @yyutiw @alyson647

Reply:
I'm not too sure about the idea to be honest. I viewed CollaborativeTask as separate to Test, Assignment, Event, and that it was its own thing since in our CRC cards it seemed like that. So, I don't know but I think that might complicate things a bit if we decide to make other entities like CollaborativeEvent, etc. It would also impact the functionality of my use case since I heavily rely on CollaborativeTask.

Reply:
@sshmuylovich @alyson647 Yeah, I'd say collaborative tasks could be something like "work on assignment" or "study for test" anyways so splitting them up further would probably overcomplicate things